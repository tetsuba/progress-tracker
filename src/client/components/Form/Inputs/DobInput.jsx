import * as React from 'react'
import { Form, Col } from 'react-bootstrap'
import { getDates, getYears } from '../utils/inputs-utils'

// TYPES
import type { eventType } from '../../../types/eventType'

type inputTypes = {
  firstName: string,
  lastName: string,
  DOBDate: string,
  DOBMonth: string,
  DOBYear: number,
  teacherID: string,
}

type Props = {
  inputs: inputTypes,
  setInputs: (e: eventType) => void,
}

export default function DobInput(props: Props) {
  const { inputs, setInputs } = props
  return (
    <Form.Row>
      <Form.Group as={Col} controlId="formGridDOB">
        <Form.Control
          as="select"
          size="lg"
          custom
          onChange={setInputs}
          value={inputs.DOBDate}
          name={'DOBDate'}
        >
          <option disabled>Date:</option>
          {getDates(inputs.DOBMonth).map((date) => (
            <option key={`dobdate-${date}`}>{date}</option>
          ))}
        </Form.Control>
      </Form.Group>
      <h3 className="text-muted p-1">/</h3>
      <Form.Group as={Col} controlId="formGridDOB">
        <Form.Control
          as="select"
          size="lg"
          custom
          onChange={setInputs}
          value={inputs.DOBMonth}
          name={'DOBMonth'}
        >
          <option disabled>Month:</option>
          <option>01</option>
          <option>02</option>
          <option>03</option>
          <option>04</option>
          <option>05</option>
          <option>06</option>
          <option>07</option>
          <option>08</option>
          <option>09</option>
          <option>10</option>
          <option>11</option>
          <option>12</option>
        </Form.Control>
      </Form.Group>
      <h3 className="text-muted p-1">/</h3>
      <Form.Group as={Col} controlId="formGridDOB">
        <Form.Control
          as="select"
          size="lg"
          custom
          onChange={setInputs}
          value={inputs.DOBYear}
          name={'DOBYear'}
        >
          <option disabled>Year:</option>
          {getYears(inputs.DOBYear).map((year) => (
            <option key={`dobyear-${year}`}>{year}</option>
          ))}
        </Form.Control>
      </Form.Group>
    </Form.Row>
  )
}
