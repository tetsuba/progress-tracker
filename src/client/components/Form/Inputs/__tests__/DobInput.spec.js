import { testRenderer } from '../../../../../test/testHelper'
import DobInput from '../DobInput'

describe('<DobInput>', () => {
  const baseProps = {
    setInputs: jest.fn(),
    inputs: {
      DOBDate: '02',
      DOBMonth: '03',
      DOBYear: 2020,
    },
  }

  describe('@Render', () => {
    it('should render', () => {
      const props = {
        ...baseProps,
      }
      const wrapper = testRenderer(DobInput, props)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
