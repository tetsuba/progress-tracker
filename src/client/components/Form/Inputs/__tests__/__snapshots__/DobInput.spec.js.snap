// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DobInput> @Render should render 1`] = `
<FormRow>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    controlId="formGridDOB"
  >
    <FormControl
      as="select"
      custom={true}
      name="DOBDate"
      onChange={[MockFunction]}
      size="lg"
      value="02"
    >
      <option
        disabled={true}
      >
        Date:
      </option>
      <option
        key="dobdate-01"
      >
        01
      </option>
      <option
        key="dobdate-02"
      >
        02
      </option>
      <option
        key="dobdate-03"
      >
        03
      </option>
      <option
        key="dobdate-04"
      >
        04
      </option>
      <option
        key="dobdate-05"
      >
        05
      </option>
      <option
        key="dobdate-06"
      >
        06
      </option>
      <option
        key="dobdate-07"
      >
        07
      </option>
      <option
        key="dobdate-08"
      >
        08
      </option>
      <option
        key="dobdate-09"
      >
        09
      </option>
      <option
        key="dobdate-10"
      >
        10
      </option>
      <option
        key="dobdate-11"
      >
        11
      </option>
      <option
        key="dobdate-12"
      >
        12
      </option>
      <option
        key="dobdate-13"
      >
        13
      </option>
      <option
        key="dobdate-14"
      >
        14
      </option>
      <option
        key="dobdate-15"
      >
        15
      </option>
      <option
        key="dobdate-16"
      >
        16
      </option>
      <option
        key="dobdate-17"
      >
        17
      </option>
      <option
        key="dobdate-18"
      >
        18
      </option>
      <option
        key="dobdate-19"
      >
        19
      </option>
      <option
        key="dobdate-20"
      >
        20
      </option>
      <option
        key="dobdate-21"
      >
        21
      </option>
      <option
        key="dobdate-22"
      >
        22
      </option>
      <option
        key="dobdate-23"
      >
        23
      </option>
      <option
        key="dobdate-24"
      >
        24
      </option>
      <option
        key="dobdate-25"
      >
        25
      </option>
      <option
        key="dobdate-26"
      >
        26
      </option>
      <option
        key="dobdate-27"
      >
        27
      </option>
      <option
        key="dobdate-28"
      >
        28
      </option>
      <option
        key="dobdate-29"
      >
        29
      </option>
      <option
        key="dobdate-30"
      >
        30
      </option>
      <option
        key="dobdate-31"
      >
        31
      </option>
    </FormControl>
  </FormGroup>
  <h3
    className="text-muted p-1"
  >
    /
  </h3>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    controlId="formGridDOB"
  >
    <FormControl
      as="select"
      custom={true}
      name="DOBMonth"
      onChange={[MockFunction]}
      size="lg"
      value="03"
    >
      <option
        disabled={true}
      >
        Month:
      </option>
      <option>
        01
      </option>
      <option>
        02
      </option>
      <option>
        03
      </option>
      <option>
        04
      </option>
      <option>
        05
      </option>
      <option>
        06
      </option>
      <option>
        07
      </option>
      <option>
        08
      </option>
      <option>
        09
      </option>
      <option>
        10
      </option>
      <option>
        11
      </option>
      <option>
        12
      </option>
    </FormControl>
  </FormGroup>
  <h3
    className="text-muted p-1"
  >
    /
  </h3>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    controlId="formGridDOB"
  >
    <FormControl
      as="select"
      custom={true}
      name="DOBYear"
      onChange={[MockFunction]}
      size="lg"
      value={2020}
    >
      <option
        disabled={true}
      >
        Year:
      </option>
      <option
        key="dobyear-2020"
      >
        2020
      </option>
      <option
        key="dobyear-2019"
      >
        2019
      </option>
      <option
        key="dobyear-2018"
      >
        2018
      </option>
      <option
        key="dobyear-2017"
      >
        2017
      </option>
      <option
        key="dobyear-2016"
      >
        2016
      </option>
      <option
        key="dobyear-2015"
      >
        2015
      </option>
      <option
        key="dobyear-2014"
      >
        2014
      </option>
      <option
        key="dobyear-2013"
      >
        2013
      </option>
      <option
        key="dobyear-2012"
      >
        2012
      </option>
      <option
        key="dobyear-2011"
      >
        2011
      </option>
      <option
        key="dobyear-2010"
      >
        2010
      </option>
      <option
        key="dobyear-2009"
      >
        2009
      </option>
      <option
        key="dobyear-2008"
      >
        2008
      </option>
      <option
        key="dobyear-2007"
      >
        2007
      </option>
      <option
        key="dobyear-2006"
      >
        2006
      </option>
      <option
        key="dobyear-2005"
      >
        2005
      </option>
    </FormControl>
  </FormGroup>
</FormRow>
`;
